zerver/lib/markdown/__init__.py

```diff
@@ -2083,15 +2083,19 @@ def make_md_engine(realm_filters_key: int, email_gateway: bool) -> None:
 # function on the URLs; they are expected to be HTML-escaped when
 # rendered by clients (just as links rendered into message bodies
 # are validated and escaped inside `url_to_a`).
-def topic_links(realm_filters_key: int, topic_name: str) -> List[str]:
-    matches: List[str] = []
+def topic_links(realm_filters_key: int, topic_name: str) -> List[Dict[str, Union[str, int]]]:
+    matches: List[Dict[str, Union[str, int]]] = []
 
     realm_filters = realm_filters_for_realm(realm_filters_key)
 
     for realm_filter in realm_filters:
         pattern = prepare_realm_pattern(realm_filter[0])
         for m in re.finditer(pattern, topic_name):
-            matches += [realm_filter[1] % m.groupdict()]
+            matched_details = m.groupdict()
```

Comment:
minor: `match_details` sounds more natural to me, even more so because we're already iterating over a match object (`for m in re.finditer()`).